### DISCUSSION TITLE
Question regarding QP2 website

### DISCUSSION THREAD

**Discussion by NastaMauger** [QuantumPackage/qp2#356] [Q&A] [answered]
Hello,

As you may already know from the different threads I have opened recently, I am experimenting with your software to get more comfortable with it. To do so, I am changing various parameters such as `weight_selection`.

According to the [documentation](https://quantum-package.readthedocs.io/en/master/modules/determinants.html#cmdoption-determinants-arg-weight_selection), it seems the default value is 2. However, when running the master or even the dev-stable branch, the default value in `qp_edit my_file.ezfio` after `qp_create_ezfio` is 1 for the keyword.

Is this normal, or could I be missing something? Additionally, I couldn't find much documentation on the different flavors of this keyword.

Thank you for your assistance.
Best

**Comment by scemama** [comment, marked as answer]
Hello,
yes, the documentation is outdated. The default is `1`, as said in `qp_edit`. (`qp_edit` is a more reliable documentation than the website).

For multi-state calculations, to get consistent energy differences between the states we need change the determinant selection rule such that we select determinants that are the best for multiple states altogether. So we use a weighted average the PT2 contribution for each state of the determinants, and these weights can be computed in different ways.

0: Use state-average weight given in input
1: Use 1/c0 as a weight, where c0 is the largest CI coefficient of the state
Others: Make the weight evolve such that the PT2 contribution or variance (or both) remains roughly the same among all states.

Using 2 could be a better choice to converge energy differences. If the energy difference converges slowly, you can try option 2. It is especially what we use for QMC calculations to have consistent wave functions between the states, since the energy difference at the variational level is expected to be close to the energy difference at the E+PT2 level (as the PT2s are almost equal).
